File: MedicApp/Models/Questionnaire.cs
namespace MedicApp.Models
{
    public class Questionnaire
    {
        public Guid Id { get; set; }
        public Guid Patient_RefID { get; set; }
        public DateTime ExpireDate { get; set; }
        public bool question1 { get; set; }
        public bool question2 { get; set; }
        public bool question3 { get; set; }
        public bool question4 { get; set; }
        public bool question5 { get; set; }
        public bool question6 { get; set; }
        public bool question7 { get; set; }
        public bool question8 { get; set; }
        public bool question9 { get; set; }
        public bool question10 { get; set; }
        public bool question11 { get; set; }
        public bool question12 { get; set; }
        public bool IsValid { get; set; }
        public bool IsDeleted { get; set; }
        public DateTime Creation_TimeStamp { get; set; }

        public Questionnaire()
        {
            Id = Guid.NewGuid();
            Creation_TimeStamp = DateTime.Now;
            IsDeleted = false;
            IsValid = false;
        }

        public bool IsQuestionireSigned()
        {

            if (this.question1)
            {
                return false;
            }
            if (this.question2)
            {
                return false;
            }
            if (this.question3)
            {
                return false;
            }
            if (this.question4)
            {
                return false;
            }
            if (this.question5)
            {
                return false;
            }
            if (this.question6)
            {
                return false;
            }
            if (this.question7)
            {
                return false;
            }
            if (this.question8)
            {
                return false;
            }
            if (this.question9)
            {
                return false;
            }
            if (this.question10)
            {
                return false;
            }
            if (this.question11)
            {
                return false;
            }
            if (this.question12)
            {
                return false;
            }
            return true;

        }
    }

    public class SaveQuestionnaire
    {
        public Guid Id { get; set; }
        public Guid Patient_RefID { get; set; }
        public DateTime ExpireDate { get; set; }
        public bool question1 { get; set; }
        public bool question2 { get; set; }
        public bool question3 { get; set; }
        public bool question4 { get; set; }
        public bool question5 { get; set; }
        public bool question6 { get; set; }
        public bool question7 { get; set; }
        public bool question8 { get; set; }
        public bool question9 { get; set; }
        public bool question10 { get; set; }
        public bool question11 { get; set; }
        public bool question12 { get; set; }
        public bool IsValid { get; set; }
    }
}

File: Client/src/app/model/Questionnaire.ts
export interface QuestionnaireInterface {
  id?: any;
  patient_RefID?: string;
  question1?: any;
  question2?: any;
  question3?: any;
  question4?: any;
  question5?: any;
  question6?: any;
  question7?: any;
  question8?: any;
  question9?: any;
  question10?: any;
  question11?: any;
  question12?: any;
  isValid?: any;
  isQuestionnaireValid?: any;
}

export class Questionnaire implements QuestionnaireInterface {
  id?: any;
  patient_RefID?: string;
  question1?: any;
  question2?: any;
  question3?: any;
  question4?: any;
  question5?: any;
  question6?: any;
  question7?: any;
  question8?: any;
  question9?: any;
  question10?: any;
  question11?: any;
  question12?: any;
  isValid?: any;
  isQuestionnaireValid?: any;

  constructor(obj: QuestionnaireInterface) {
    this.id = obj.id;
    this.patient_RefID = obj.patient_RefID;
    this.question1 = obj.question1;
    this.question2 = obj.question2;
    this.question3 = obj.question3;
    this.question4 = obj.question4;
    this.question5 = obj.question5;
    this.question6= obj.question6;
    this.question7 = obj.question7;
    this.question8 = obj.question8;
    this.question9 = obj.question9;
    this.question10 = obj.question10;
    this.question11 = obj.question11;
    this.question12 = obj.question12;
    this.isValid = obj.isValid;
    this.isQuestionnaireValid = obj.isQuestionnaireValid;
  }
}
